geode-core/src/main/java/org/apache/geode/cache/Region.java

```diff
@@ -178,6 +179,26 @@
    */
   <PK, PV> Region<PK, PV> getParentRegion();
 
+  /**
+   * Gets the root ancestor region for this region. If the region itself is a root region
+   * (i.e. has no parents), returns null.
+   * <p>
+   * Does not throw a <code>CacheClosedException</code> or a <code>RegionDestroyedException</code>.
+   *
+   * @return the root ancestor region which contains this region; null,
+   *         if this region is the root region
+   * @see Region#createSubregion(String, RegionAttributes) createSubregion
+   */
+  default <PK, PV> Region<PK, PV> getRootRegion() {
```

Comment:
I would not add this to Region since we want to deprecate subregions. Once subregions go away this method would always return null. Region is an external API that we need to keep supporting.